### PR TITLE
Refactor aug30

### LINKED ISSUE #32: Add .gitignore to template output
It would be sweet when you create a new project via `mint new` that it would create a .gitignore file that's based on the template you selected. 

e.g. it would know the name of the build directory based on the template you selected, and include that in the .gitignore file, along with other commonly ignored files and folders like `.cache`. 

I could see a lot of devs accidentally committing their node_modules folder by accident.

### PR DESCRIPTION
**Pull request by KmBarnett**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Feature
- [x] Bug Fix
- [x] Optimization
- [x] Documentation Update

## Description

This PR Is for a major refactor of the CLI 
it adds org support as well as switches syntax to es6
Should address the .gitignore file copying

## Added to documentation?

- [x] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -40,7 +40,7 @@ or for the cool kids:
 ```shell
 mint n <my-project>
 cd <my-project>
-mint r -cp
+mint r
```
Comment (monarchwadia) [inline README.md:55]: I honestly feel like we should just have this info in the `--help` command. What do you think?

**Review by monarchwadia — DISMISSED** [review]

File: README.md (hunk 1 above)
Comment (KmBarnett) [reply README.md:55]: Maybe both I like it in the README

**Review by clairefro — COMMENTED** [review]
see comments on PR #35 (this refactor is included there)

**Comment by clairefro** [comment]
Closing as this is included in #35

**Pull request closed without merging**